ote/test/clj/ote/transit_changes/detection_test.clj

```diff
@@ -513,17 +550,87 @@
     (testing "second change date is correct"
       (is (= (d 2019 6 3) (-> diff-maps second :different-week :beginning-of-week))))))
 
+(def data-change-to-summer-schedule-nil-weeks-with-midsummer
+  (weeks (d 2019 5 13)
+         {route-name ["1A" "1A" "1A" "1A" "1A" "6A" "7A"]} ;; 2019 05 13
+         {route-name ["1A" "1A" "1A" "1A" "1A" "6A" "7A"]} ;; 2019 05 20
+         {route-name ["1A" "1A" "1A" "7A" "5B" "6A" "7B"]} ;; 2019 05 27
+         {route-name [nil nil nil nil nil "B6" nil]}       ;; 2019 06 03
+         {route-name [nil nil nil nil nil "B6" nil]}       ;; 2019 06 10
+         {route-name [nil nil nil nil "C5" nil nil]}       ;; 2019 06 17
+         {route-name [nil nil nil nil nil "B6" nil]}       ;; 2019 06 24
+         {route-name [nil nil nil nil nil "B6" nil]}
+         {route-name [nil nil nil nil nil "B6" nil]}
+         {route-name [nil nil nil nil nil "B6" nil]}
+         {route-name [nil nil nil nil "C5" "B6" nil]}
+         {route-name [nil nil nil nil nil "B6" nil]}
+         {route-name [nil nil nil nil nil "B6" nil]}))
+
+(def data-traffic-winter-to-summer-and-end-traffic
+  (weeks (d 2019 5 13)
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}  ;; 2019 05 13
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}  ;; 2019 05 20
+         {route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]}  ;; 2019 05 27
+         {route-name [nil nil nil nil nil "h6" nil]}        ;; 2019 06 03
+         {route-name [nil nil nil nil nil "h6" nil]}        ;; 2019 06 10
+         {route-name [nil nil nil nil nil "h6" nil]}        ;; 2019 06 17
+         {route-name [nil nil nil nil nil nil nil]}         ;; 2019 06 24
+         {route-name [nil nil nil nil nil nil nil]}
+         {route-name [nil nil nil nil nil nil nil]}
+         {route-name [nil nil nil nil nil nil nil]}))
+
+(deftest test-traffic-winter-to-summer-and-end-traffic
+  (let [result (-> data-traffic-winter-to-summer-and-end-traffic
+                   detection/changes-by-week->changes-by-route
+                   detection/detect-changes-for-all-routes)]
+    (is (= {:no-traffic-start-date (d 2019 6 23)
+            :route-key route-name
+            :starting-week {:beginning-of-week (d 2019 5 20) :end-of-week (d 2019 5 26)}}
+           (select-keys (nth result 0)
+                        [:no-traffic-start-date :no-traffic-end-date :route-key :no-traffic-change
+                         :different-week :route-key :starting-week])))
+    (is (= 1 (count result)))))
```

Comment:
When route end is fixed in your next pr, this will fail.

Reply:
Fixed in next PR